test/reviewController.test.js

```diff
@@ -124,4 +124,95 @@ describe('ReviewController API Tests', () => {
       expect(response.body.allProductReviews).toEqual([]);
     }
   });
+  //create a review
+  it('should create a review successfully', async () => {
+    await Review.deleteMany({});
+    const userCredentials = {
+      email: 'Emily@google.com',
+      password: 'secret',
+    };
+
+    const signedCookie = await loginAndReturnCookie(userCredentials);
+
+    const album = await Album.findOne({ albumName: 'Unique Album' });
+
+    const newReviewData = {
+      rating: 4,
+      title: 'Loved this review',
+      comment: 'Recommend it"',
+      user: user._id,
+      album: album._id,
+    };
+    const response = await request(app)
+      .post(`/api/v1/reviews/album/${album._id}`)
+      .set('Cookie', signedCookie)
+      .send(newReviewData);
+
+    expect(response.status).toBe(StatusCodes.CREATED);
+    expect(response.body).toHaveProperty('review');
+    // If no album was found
+    if (!album) {
+      expect(response.status).toBe(StatusCodes.NOT_FOUND);
+    }
+    //create second review for the same album
+    const secondReviewData = {
+      rating: 4,
+      title: 'Loved this review',
+      comment: 'Recommend it"',
+      user: user._id,
+      album: album._id,
+    };
+    const res = await request(app)
+      .post(`/api/v1/reviews/album/${album._id}`)
+      .set('Cookie', signedCookie)
+      .send(secondReviewData);
+    // this user already submitted a review for this album
+    expect(res.status).toBe(StatusCodes.CONFLICT);
+  });
+
+  //  update a review
+  it('should update a review successfully', async () => {
+    const userCredentials = {
+      email: 'Emily@google.com',
+      password: 'secret',
+    };
+
+    const signedCookie = await loginAndReturnCookie(userCredentials);
+
+    const existingReview = await Review.findOne({});
+    // If no existingReview was found
+    if (!existingReview) {
+      expect(response.status).toBe(StatusCodes.NOT_FOUND);
+    }
```

Comment:
I generally don't expect to see much conditional logic in a test. Each test should be testing one, _maybe two_ specific things. It's better to have many small tests, than one long test that is trying to check too many diverging things at once. 

In this case, we just fetched a review from the db at line 182. We _know_ that a review exists since we created it in the `beforeEach` block and we haven't deleted it in this test. So we _know_ that there will always be an existing review. So there's no reason to check _if_ there's no existing review. In test code, we for the most part can control everything about the state of the "world" that we're testing. So we shouldn't need to check multiple branches/optional code paths, instead we should be "Arranging" that things go down a certain path and then "Asserting" that the result of that path is what we expect it to be.

In this case, it would be better instead to have a separate test that asserts the `NOT_FOUND` result.

Reply:
@akosasante 
done, thank you so much!!